components/tutorial/tutorial_tip/index.ts

```diff
@@ -4,23 +4,36 @@ import {connect} from 'react-redux';
 import {bindActionCreators, Dispatch} from 'redux';
 
 import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
-import {getAutoTourTreatment, getInt} from 'mattermost-redux/selectors/entities/preferences';
+import {
+    getAutoTourTreatment,
+    getInt,
+    makeGetCategory,
+} from 'mattermost-redux/selectors/entities/preferences';
 import {savePreferences} from 'mattermost-redux/actions/preferences';
 import {GenericAction} from 'mattermost-redux/types/actions';
+import {PreferenceType} from 'mattermost-redux/types/preferences';
 import {AutoTourTreatments} from 'mattermost-redux/constants/config';
 
 import {closeMenu as closeRhsMenu} from 'actions/views/rhs';
-import {Preferences} from 'utils/constants';
+import {setFirstChannelName} from 'actions/views/channel_sidebar';
+
+import {Preferences, RecommendedNextSteps} from 'utils/constants';
 import {GlobalState} from 'types/store';
 
 import TutorialTip from './tutorial_tip';
 
 function mapStateToProps(state: GlobalState) {
+    const getCategory = makeGetCategory();
     const currentUserId = getCurrentUserId(state);
+    const preferences = getCategory(state, Preferences.AB_TEST_PREFERENCE_VALUE);
+    const firstChannelNameFromPref = preferences.find((pref: PreferenceType) => pref.name === RecommendedNextSteps.CREATE_FIRST_CHANNEL);
+    const firstChannelNameFromRedux = state.views.channelSidebar.firstChannelName;
+    const currentStep = firstChannelNameFromRedux || firstChannelNameFromPref?.value ? -1 : getInt(state, Preferences.TUTORIAL_STEP, currentUserId, 0);
```

Comment:
Here and in `components/sidebar/sidebar_channel/sidebar_channel_link/index.ts`, I think setting to `-1` based on first channel names should be passed as the last argument to `getInt` instead of overriding all the time. I need to check on a fresh setup, but I think for first admins, they will get stuck on seeing the first tip until they refresh the screen. In addition, if they click more than once on the same tip, I think they could accidentally skip tips.

I'm thinking something like

```ts
let defaultStep = TutorialSteps.POST_POPOVER;
if (firstChannelNameFromRedux || firstChannelNameFromPref?.val) {
  defaultStep = TutorialSteps.ADD_FIRST_CHANNEL;
}
const currentStep = getInt(state, Preferences.TUTORIAL_STEP, currentUserId, defaultStep);
```

Note that for sidebar_channel_link/index.tx, this would need to be adjusted so that the default step when there is no first channel name is `TutorialSteps.FINISHED`.

Reply:
I added the implementation we discuss internally of saving the `tutorial_step` value to `-1 `into the preferences always when the guided first channel is created.

Reply:
LGTM now! :shipit: :partying_face: